### PR TITLE
Nate use case 4

### PR DESCRIPTION
**Pull request by nathanfrommpls**

### CODE REVIEW

File: src/store/entities/Store.java [hunk 1]
```diff
@@ -56,8 +56,53 @@ public Product addProduct(String id, String name, double price, int inventory, i
...
 
-	public String checkout(Hashtable<String, Integer> cart) {
-		return "Checkout Output.";
+	public String checkout(ArrayList<String> productIds,
+	ArrayList<Integer> productQtys) {
+		double runningTotal = 0;
+		StringBuilder buffer = new StringBuilder();
+		ArrayList<String> reorderList = new ArrayList<String>();
+		
+		if ( productIds.size() != productQtys.size() ) {
```
Comment (DanielRenaudMN) [inline src/store/entities/Store.java:65]: Not sure about this one

**Review by DanielRenaudMN — CHANGES_REQUESTED** [review]
I've got questions about your liberal use of buffer.append. I suspect the code is more complicated than it needs to be. Let the power of variables compel you! They improve code readability. Smaller changes I recommend:

-Strings will be read by the user. Therefore it is best to translate your messages to more common English. See lines 60, 66, and 228.

-Grammar fixes are needed but I'll take care of that on another pass through. 

-According to Brahma's coding guide (at least as it is written), we shouldn't be using i, j, or k. We are supposed to use words for all variables.

**Review by nathanfrommpls — COMMENTED** [review]
> I've got questions about your liberal use of buffer.append. I suspect the code is more complicated than it needs to be. Let the power of variables compel you! They improve code readability. Smaller changes I recommend:

Can you expound on what you mean by "complicated"? Do you mean "has too many lines" or something else? As far as the power of variables go, I am using variables so it is unclear to me what you mean.

> Strings will be read by the user. Therefore it is best to translate your messages to more common English. See lines 60, 66, and 228.

This is a fair point. When I had started I was thinking about common separated values as something that good be parsed and displayed by the UI in a flexible way. Allowing the UI to control how the data was displayed and the store to just handle the data itself, but its probably not worth it.

> Grammar fixes are needed but I'll take care of that on another pass through.

Are you talking about English Grammar or something to do with Java? Without specifics I can't fix anything.

> According to Brahma's coding guide (at least as it is written), we shouldn't be using i, j, or k. We are supposed to use words for all variables.

You've got me there, that is technically true.

Thanks for all of the feedback.

**Comment by DanielRenaudMN** [comment]
> > I've got questions about your liberal use of buffer.append. I suspect the code is more complicated than it needs to be. Let the power of variables compel you! They improve code readability. Smaller changes I recommend:
> 
> Can you expound on what you mean by "complicated"? Do you mean "has too many lines" or something else? As far as the power of variables go, I am using variables so it is unclear to me what you mean.
> 
> > Strings will be read by the user. Therefore it is best to translate your messages to more common English. See lines 60, 66, and 228.
> 
> This is a fair point. When I had started I was thinking about common separated values as something that good be parsed and displayed by the UI in a flexible way. Allowing the UI to control how the data was displayed and the store to just handle the data itself, but its probably not worth it.
> 
> > Grammar fixes are needed but I'll take care of that on another pass through.
> 
> Are you talking about English Grammar or something to do with Java? Without specifics I can't fix anything.
> 
> > According to Brahma's coding guide (at least as it is written), we shouldn't be using i, j, or k. We are supposed to use words for all variables.
> 
> You've got me there, that is technically true.
> 
> Thanks for all of the feedback.

**Comment by DanielRenaudMN** [comment]
I'd have to play with the code to give you a more detailed remark, but I suspect that there are more lines than necessary and that buffer.append isn't being used correctly. If you can't simplify it, just resubmit as is and I'll take a crack at it later.

I was talking about English grammar. In general, issues with coding grammar are caught by the IDE. It's not as important to go into detail on because it's not something Brahma cares about; it just makes our work look more professional.

**Comment by nathanfrommpls** [comment]
@DanielRenaudMN @DanielRenaudMN  @atvick21 @butters008 

> I'd have to play with the code to give you a more detailed remark, but I suspect that there are more lines than necessary and that buffer.append isn't being used correctly.

As a group what standard would we like for blocking PRs? Is a suspicion, intuition, or a hunch sufficient? I'd like for us to quickly come to an agreement on this matter so that we can continue moving forward, as we don't have an abundance of time.

I would suggest that we apply the following criteria when asking evaluating PRs.
- output of a test showing a failure OR documentation showing specific code is syntactically incorrect, and when possible proposed changes
- focus on whether something provides correct output not whether something is implemented the way we would have implemented it
- if something is broken, "fix forward" rather than trying to undo work

More specifically, this PR doesn't affect anything outside ui.checkout() and store.checkout() in any meaningful way, so even it is terribly broken, it is not breaking anything else.

**Comment by DanielRenaudMN** [comment]
That seems unnecessary. This isn't taking up much time. As I said, you should make the changes I brought up that you think are valid (you didn't dispute all my remarks, therefore my review was valid). I can propose additional changes at a later time that you can then review.

**Review by raelkeru — APPROVED** [review]
I agree with Daniel about using common English for stuff being displayed to the user. These are mainly preferences though, so I think we should proceed with this for the time being. Everything looks functional from my review, so I'm going to go ahead and approve it so we're all looking at the most up-to-date version.

**Pull request merged**